wit/couchbase/fts.wit

```diff
@@ -0,0 +1,216 @@
+///   - Dates
+///   - Numbers
+///   - CIDR notation
+///   - Geospatial data
+///
+/// Reference: https://docs.couchbase.com/server/current/search/search.html
+interface fts {
+  use types.{
+    request-span, document, document-field-name, collection-name, sort-direction, 
+    bucket-error, search-index-name, retry-strategy
+    };
+
+  variant fts-bucket-error {
+    /// A completely unexpected error
+    unexpected(string),
+  }
+
+  /// Buckets for use with FTS
+  ///
+  /// NOTE: This is unnecessary (and may be implemented as a no-op) for Couchbase versions greater than 6.5
+  resource fts-bucket {
+    constructor(name: string);
+    /// Wait until a bucket is ready to use
+    ///
+    /// This should be done on couchbase versions earlier than 6.5, before performing any search queries.
+    wait-until-ready: func() -> result<_, bucket-error>;
+  }
+
+  /// Errors that occur when creating an FTS index
+  variant fts-index-create-error {
+    /// A completely unexpected error
+    unexpected(string)
+  }
+
+  /// Errors that occur when creating an FTS index
+  variant fts-bucket-create-error {
+    /// A completely unexpected error
+    unexpected(string)
+  }
+
+  /// Errors that occur when performing an FTS search
+  variant fts-search-error {
+    /// A completely unexpected error
+    unexpected(string)
+  }
+
+  /// Search facet
+  variant facet {
+    /// Term Facet
+    term(tuple<string, u64>),
+    /// Numeric Facet
+    numeric(tuple<string, u64>),
+    /// Date range
+    date-range(tuple<string, string, string>),
+  }
+
+  /// Collection of mutation tokens that represent a single state
+  record mutation-state {
+    tokens: list<mutation-token>,
+  }
+
+  /// Individual mutation token
+  record mutation-token {
+    bucket-name: string,
+    partition-uuid: u64,
+    partition-id: u64,
+    sequence-number: u64,
+  }
+
+  /// Options to use when performing a search
+  record search-options {
+    /// Number of results to limit the search results by
+    limit: u32,
+
+    /// Level of data consistency desired for a search query
+    scan-consistency: u64,
+
+    /// Number of results to skip
+    skip: u32,
+
+    /// Enable explain on the search
+    explain: bool,
+
+    /// Enable highlighting on the search
+    highlight: bool,
+
+    /// Fields on documents in which to search
+    fields: list<document-field-name>,
+
+    // Fields to sort by
+    sort: list<tuple<document-field-name, sort-direction>>,
+
+    facets: list<tuple<document-field-name, facet>>,
+    consistent-with: mutation-state,
+
+    /// Whether to disable scoring
+    disable-scoring: bool,
+
+    /// Milliseconds until the search operation should time out
+    timeout-ms: u64,
+
+    /// How and whether to retry the operation
+    retry-strategy: option<retry-strategy>,
+
+    /// A known span to associate this lookup with
+    parent-span: option<request-span>,
+
+    /// Collections in which to search
+    collections: list<collection-name>,
```

Comment:
`collection-name` is defined as a `string`. That creates an ambiguity between the simple _name_ (i.e.: `_default`)and the _keyspace path_ (i.e.: `myBucket._default._default`).

Since there is no `query-context` option, I assume the keyspace path is expected. Do we just do with solid documentation or do we alleviate the ambiguity by requesting an object of the shape `{ bucket, scope, collection }` ?

Reply:
I personally like the idea of being explicit with `bucket`, `scope` and `collection` (maybe `name`, since it will be a `record collection`)

I'll add this!

- [ ] More explicit `collection` type